jobrunner/manage_jobs.py

```diff
@@ -340,12 +336,25 @@ def finalise_job(job):
                 copy_file(workspace_dir / filename, medium_privacy_dir / filename)
                 new_files.append(filename)
         delete_files(medium_privacy_dir, existing_files, files_to_keep=new_files)
-        write_manifest_file(medium_privacy_dir, manifest)
 
-    # Don't update the primary manifest until after we've deleted old files
-    # from both the high and medium privacy directories, else we risk losing
-    # track of old files if we get interrupted
-    write_manifest_file(workspace_dir, manifest)
+        # osrelease needs to be able to read the workspace name and repo URL from somewhere, in order to avoid the
+        # person doing the release having to enter all the details. So we write this rump manifest just into the
+        # medium privacy workspace. release-hatch is launched with this information already provided, so when osrelease
+        # has been removed we can stop doing this.
+        #
+        # We only really need to write this the first time that an action is run in a workspace, but it's easier to do
+        # it here every time than try to detect that. Writing this here every time also ensures that we get rid of old
+        # versions of the manifest that have no-longer-used-and-soon-to-be-out-of-date data in them (see the comment
+        # below).
+        write_manifest_file(
+            medium_privacy_dir, {"repo": job.repo_url, "workspace": job.workspace}
+        )
+
+    # We no longer use the manifest file in the high privacy workspace. Delete any old instances of it so that we're
+    # not left with an out-of-date version sitting around.
+    manifest_file = workspace_dir / METADATA_DIR / MANIFEST_FILE
+    if manifest_file.exists():
+        manifest_file.unlink()
```

Comment:
On windows, `unlink()` has a non-negligable chance of failing, so it might we worth adding exception handling

Reply:
We don't have any equivalent error handling here: https://github.com/opensafely-core/job-runner/blob/master/jobrunner/manage_jobs.py#L517.

I guess we just let the exception bubble up to the `catch Exception` clause in `run.py`.

Reply:
Ok. Given it's not a fatal failure, I figured it was worth suppressing any error, but not that bothered by it.